xmppserver/src/main/java/org/jivesoftware/openfire/session/LocalClientSession.java

```diff
@@ -945,6 +945,11 @@ public boolean isMessageCarbonsEnabled() {
     @Override
     public void setMessageCarbonsEnabled(boolean enabled) {
         messageCarbonsEnabled = enabled;
+        if (ClusterManager.isClusteringStarted()) {
+            // Track information about the session and share it with other cluster nodes
+            Cache<String,ClientSessionInfo> cache = SessionManager.getInstance().getSessionInfoCache();
+            cache.put(getAddress().toString(), new ClientSessionInfo(this));
```

Comment:
Why is it needed to modify the caches here (and on line 966)? It is strange to me that when these new properties get updated, a cache entry is added/updated, but when the existing properties get updated, this is not done.

Reply:
My previous remark was incorrect. The exact same thing _does_ happen, for example in `org.jivesoftware.openfire.session.LocalClientSession#setOfflineFloodStopped`